Limit threads to 50 to 100 range.

rtibbles opened the pull request:
### Summary
Follow up from #5296 to limit threads to a range of 50 to 100.

----

### Contributor Checklist


PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.rst

indirectlylit commented on kolibri/utils/options.py:
```diff
@@ -22,14 +22,19 @@ def calculate_thread_pool():
-            pool = int(45 * total_memory - 80)  # 10 for 2 Gb,100 for 4 Gb
-            pool = 200 if pool > 200 else pool
+        if min_mem < total_memory < max_mem:
+            pool = int(MAX_POOL * float(total_memory) / max_mem)
```
This I think works because `MIN_POOL / MAX_POOL` currently equals `min_mem / max_mem` but I believe it may not work as expected otherwise. perhaps something like:

```python
MIN_POOL + (MAX_POOL - MIN_POOL) * float(total_memory) / max_mem
```

rtibbles replied:
Yes, can update.

rtibbles committed:
Limit threads to 50 to 100 range.

jredrejo reviewed (commented):
In the tests we did in Honduras, with 8 Gb of RAM, 150 was working better than 100.
I would not limit it to 100, it's too low from my pov. I had not tested 200, but 150 worked better than 100 if the server can handle it.

indirectlylit committed:
updates to thread calculations

- use multiple return statements instead of updating a local variable
- clean up comments
- increase max thread count to 150

indirectlylit commented:
pushed commit cbf60605015838e205095f875097c840b58fcbb6

- increase max thread count to 150
- use multiple return statements instead of updating a local variable
- clean up comments

indirectlylit committed:
bump max from 4 to 6GB
